### PR TITLE
Change the Button component background and text colour

### PR DESCRIPTION
**Pull request by NickColley**
Expands the public API to include a new `$govuk-button-background-colour` and `$govuk-button-text-colour` scss variable.

Allows users who brand GOV.UK to do so without brittle selector overrides
which can break in future updates.

This variable naming matches other similar variables such as
`$govuk-body-background-colour` and `$govuk-canvas-background-colour`.

At the moment I have to do this sort of thing:

```scss
@import "node_modules/govuk-frontend/govuk/base";
@import "node_modules/govuk-frontend/govuk/components/button/index";

$app-button-colour: $app-primary-color;
$app-button-hover-colour: govuk-shade($app-button-colour, 20%);
$app-button-shadow-colour: govuk-shade($app-button-colour, 60%);
.govuk-button:not(.govuk-button--secondary):not(.govuk-button--warning) {
    background-color: $app-button-colour;
    box-shadow: 0 $govuk-border-width-form-element 0 $app-button-shadow-colour;
}

.govuk-button:not(.govuk-button--secondary):not(.govuk-button--warning):hover {
    background-color: $app-button-hover-colour;
}
```

After:

```scss
@import "node_modules/govuk-frontend/govuk/base";

$govuk-button-background-colour: govuk-colour("blue");
@import "node_modules/govuk-frontend/govuk/components/button/index";
```

### CODE REVIEW

**Comment by NickColley** [comment]
Is it possible to get this considered for triage please?

@36degrees @vanitabarrett

**Comment by NickColley** [comment]
Thank you @vanitabarrett ! Appreciate it.

File: src/govuk/components/button/_index.scss
Comment (querkmachine) [inline src/govuk/components/button/_index.scss:10]: Would this line be better placed in `src/govuk/settings/_colours-applied.scss`? That's where other default variables definitions seem to be located. 

I'm conscious that this variable is specific to the button component, however, and we don't seem to have precedent for where to put them specifically. Happy to keep it here if we think that makes more sense.

File: src/govuk/components/button/_index.scss
Comment (querkmachine) [inline src/govuk/components/button/_index.scss:16]: As we're allowing customisation of the background-color, it feels like we should also provide a mechanism to customise the text color so that it's possible to maintain accessible contrast.

**Review by querkmachine — COMMENTED** [review]
Couple of thoughts. Nothing explicitly blocking.

File: src/govuk/components/button/_index.scss
Comment (NickColley) [reply src/govuk/components/button/_index.scss:10]: I'm thinking over time I can contribute the different high level component variables based on me branding the GOV.UK Design System so it'll make more sense.

File: src/govuk/components/button/_index.scss
Comment (NickColley) [reply src/govuk/components/button/_index.scss:16]: Agreed, this'll also be useful when either I or the GOV.UK Design System team writes about branding. My plan is to contribute these improvements based on real world usage of branding while doing client work. Then writing guidance on how this is done that could be contributed back to the main GOV.UK Design System if it becomes useful to do so.

**Comment by NickColley** [comment]
@querkmachine pushed an update addressing your feedback.

**Comment by querkmachine** [comment]
A commit from your accordion focus colour PR seems to have snuck in and caused a merge conflict. If you can fix that up, I'm happy to approve!

**Comment by NickColley** [comment]
Ah yeah rouge rebase gone wrong, will sort that thanks.

**Comment by querkmachine** [comment]
Hi @NickColley,

Sorry! This has ended up overlapping with some other work we're doing and we've inadvertently introduced a new merge conflict to your PR — the file at `tasks/gulp/sassdoc.js` has been moved to `tasks/sassdoc.js` on the main branch now. 

If you're okay to take a look at it, please do. We're currently preparing a bugfix release of Frontend however, so it may be a few more days until this is approved and merged.

**Comment by NickColley** [comment]
@querkmachine okay sorted that conflict thanks for the heads up. 👍🏻

**Comment by querkmachine** [comment]
@NickColley I've just rebased your branch and moved the changelog entry to the right place now that the 4.3.1 release has gone out. Will approve once the tests have ran. Thanks for putting up with the faff!

**Review by querkmachine — APPROVED** [review]

**Review by owenatgov — APPROVED** [review]
🟢 🔴 🟡 👍🏻

**Comment by NickColley** [comment]
Thanks @querkmachine @owenatgov for your help getting this one in 👍🏻

**Pull request merged**
